Http server

HanYJune opened the pull request.

HanYJune committed:
test

HanYJune committed:
start

HanYJune committed:
start

HanYJune committed:
add gitignore

HanYJune committed:
add gitignore

HanYJune committed:
add gitignore

HanYJune committed:
Delete http_server/node_modules directory

kevhahn97 commented on http_server/src/routes/CLI.js:
```diff
@@ -0,0 +1,8 @@
+const express = require('express');
+const router = express.Router();
+
+router.post('/get_refactoring_code', );
```
1. cli랑 web이랑 endpoint를 분리할 필요가 있을까 싶은데 영석님이랑 합의된 부분인가요?
제가 원래 이해하기로는 기능(탄소배출량 측정, 리팩토링)별로 endpoint 하나씩 제공하면 공유할 수 있다는 거여서

2. status request도 합의된 부분인가요? firebase db 특성상 프론트엔드에서 바로 접근할 수 있기도 해서 꼭 http server를 거칠 필요는 없긴해요

HanYJune replied:
1. 아녀 그부분 얘기하다가 말아서 아직 안정해졌어요. web이랑 cli에서 코드 읽어오는 방식을 다르게 해야하나<< 이거를 아직 안정해서요. 디렉토리 읽는거를 구현해본적이 없어서 어떻게 될지 상상이 안되는데 분리안해도 상관 없으려나요 
2.  그럼 빼는게 좋아보이네요. 이거 영석님이랑 합의된 부분이긴한데 영석님도 firebase db 프론트에서 바로 접근하는 거를 인지를 못하신거 같아요

kevhahn97 replied:
1 -> 문서에 댓글 남겼지만 일단 코드 파일 = utf8 string 으로 약속하고 interface 통일하는 쪽으로
2 -> 일단 빼는거로

지금은 틀만 잡는거니까 이거만 수정하고 노티 주시면 approve 드릴게요~

HanYJune replied:
넵 내일 회의전에 올리겠습니다

HanYJune replied:
job producer 와 job 을 모델객체로 두고 컨트롤러에서 job producer를 불러서 job을 생성하고 enqueue,update storage,update db를 수행하게끔 코드를 짰는데, (controllers/CodeController 부분)
이 방식보다 컨트롤러자체를 job producer로 두는게 이쁠까요? 이렇게 하면job 객체안에 enqueue, update storage, update db로 두고 job producer 컨트롤러가 new job(), job.enqueue  job.update_storage 이런식으로 구현됩니다.

kevhahn97 replied:
문서에서는 전자로 하지 않앗어요? 크게 상관없을거 같고 그냥 전자로 가시죠

HanYJune replied:
네 문서에는 전자로 했어요. 그럼 그냥 전자로 가겠습니다

HanYJune replied:
푸시했어요

HanYJune replied:
웹에서 파일받고 그린서버로 보내고 받는거 까지 됐습니다.

HanYJune replied:
firebase: 팀용 db말고 test용 db 따로 만들어서 연동 확인한 코드 입니다

HanYJune committed:
integrate cli and web + delete get status code

HanYJune committed:
Merge branch 'main' of https://github.com/skkuse/2024spring_41class_team9 into http_server

HanYJune committed:
Merge branch 'http_server' of https://github.com/skkuse/2024spring_41class_team9 into http_server

HanYJune committed:
cli web -> integrate & delete job status

HanYJune committed:
integrate cli, web & delete status

HanYJune committed:
test

HanYJune committed:
test

HanYJune committed:
test

HanYJune committed:
interface:web&cli,green server

firebase connected

completed except enqueue

TODO: ENQUEUE

aeng?

enqueue finish

enqueue finish

HanYJune committed:
interface:web&cli,green server

HanYJune committed:
firebase connected

kevhahn97 commented on http_server/src/controllers/CodeController.js:
```diff
@@ -0,0 +1,60 @@
+};
+
+exports.sendToMeasure = async (req, res, next) => {
+  // id 발행해야 함
```
id 발행은 firestore에서 document 생성하면 임의의 uuid 하나 발행해 줄거예요. 그거 그대로 사용해도 좋을듯요

kevhahn97 commented on http_server/src/controllers/CodeController.js:
```diff
@@ -0,0 +1,60 @@
+  const files = req.body.files;
+  const numOfFiles = files.length;
+
+  res.send("ok");
```
이건 뭔가요?

HanYJune replied:
11 13 번 줄 코드는 그냥 테스트용으로 해본건데 무시하셔도 됩니다

HanYJune committed:
completed except enqueue

TODO: ENQUEUE

aeng?

HanYJune committed:
enqueue finish

HanYJune committed:
enqueue finish

HanYJune committed:
delete env and add env.sample

HanYJune committed:
delete env and add env.sample

HanYJune committed:
Delete http_server/.env

HanYJune committed:
Merge branch 'http_server' of https://github.com/skkuse/2024spring_41class_team9 into http_server

HanYJune committed:
force push

HanYJune committed:
asd

HanYJune committed:
aaa

HanYJune committed:
final

kevhahn97 committed:
Merge branch 'main' into http_server

kevhahn97 committed:
feat(http_server): allow CORS

kevhahn97 committed:
feat(http_server): return job_id
